app/scripts/fetch_xulrunner

```diff
@@ -266,6 +270,29 @@ function modify_omni {
 	replace_line 'AddonManagerPrivate.callAddonListeners' 'await AddonManagerPrivate.callAddonListeners' modules/addons/XPIInstall.sys.mjs
 	replace_line 'let uninstall = \(\) => \{' 'let uninstall = async () => {' modules/addons/XPIInstall.sys.mjs
 	replace_line 'cancelUninstallAddon\(aAddon\)' 'async cancelUninstallAddon(aAddon)' modules/addons/XPIInstall.sys.mjs
+	# Check for `discontinue` flag in update config and uninstall the addon if set
+	replace_line 'if \(update && !this.addon.location.locked\) \{' \
+		'if (update && !this.addon.location.locked) {
+			if (update.discontinue) {
+				let aId = this.addon.id;
+				logger.debug(`AddonUpdateChecker: uninstall addon \${aId} with discontinue flag`);
+				let addon = await AddonManager.getAddonByID(aId);
+				if (addon && (addon.permissions & AddonManager.PERM_CAN_UNINSTALL)) {
+					try {
+						addon.uninstall();
+					} catch (err) {
+						logger.error(`AddonUpdateChecker: error uninstalling addon \${aId} with discontinue flag`, err);
+					}
+					sendUpdateAvailableMessages(this, null);
+					return;
+				}
+				else {
+					logger.error(`AddonUpdateChecker: addon ]${aId} with discontinue flag not found or cannot be uninstalled`);
```

Comment:
```suggestion
					logger.error(`AddonUpdateChecker: addon \${aId} with discontinue flag not found or cannot be uninstalled`);
```

I'm assuming.

But is this escaping for Perl or the shell? If it's the shell, we don't need it in a single-quote string:

```bash
$ echo '$PATH'
$PATH
$ echo '\$PATH'
\$PATH
```

Reply:
Thanks, sorry for missed that! I don’t know where it came from, but we need it here.